examples/vision/yolov8.py

```diff
@@ -0,0 +1,628 @@
+
+"""
+First we will create a instance of backbone which will be used by our yolov8 detector
+class. 
+
+YOLOV8 Backbones available in KerasCV:
+
+1.   Without Weights:
+
+    1.   yolo_v8_xs_backbone
+    2.   yolo_v8_s_backbone
+    3.   yolo_v8_m_backbone
+    4.   yolo_v8_l_backbone
+    5.   yolo_v8_xl_backbone
+
+2. With Pre-trained coco weight:
+
+    1.   yolo_v8_xs_backbone_coco
+    2.   yolo_v8_s_backbone_coco
+    2.   yolo_v8_m_backbone_coco
+    2.   yolo_v8_l_backbone_coco
+    2.   yolo_v8_xl_backbone_coco
+
+
+
+"""
+
+backbone = keras_cv.models.YOLOV8Backbone.from_preset(
+    "yolo_v8_s_backbone_coco"  # We will use yolov8 small backbone with coco weights
+)
+
+"""
+Next, let's build a YOLOV8 model using the `YOLOV8Detector`, which accepts a feature
+extractor as the `backbone` argument, a `num_classes` argument that specifies the number
+of object classes to detect based on the size of the `class_mapping` list, a
+`bounding_box_format` argument that informs the model of the format of the bbox in the
+dataset, and a finally, the feature pyramid network (FPN) depth is specified by the
+`fpn_depth` argument.
+
+It is simple to build a YOLOV8 using any of the aforementioned backbones thanks to
+KerasCV.
+
+"""
+
+yolo = keras_cv.models.YOLOV8Detector(
+    num_classes=len(class_mapping),
+    bounding_box_format="xyxy",
+    backbone=backbone,
+    fpn_depth=1,
+)
+
+"""
+## Compile the Model
+"""
+
+"""
+Loss used for YOLOV8
+
+
+1. Classification Loss: This loss function calculates the discrepancy between anticipated
+class probabilities and actual class probabilities. In this instance,
+`binary_crossentropy`, a prominent solution for binary classification issues, is
+Utilized. We Utilized binary crossentropy since each thing that is identified is either
+classed as belonging to or not belonging to a certain object class (such as a person, a
+car, etc.).
+
+2. Box Loss:  `box_loss` is the loss function used to measure the difference between the
+predicted bounding boxes and the ground truth. In this case, the Intersection over Union
+(IoU) metric is used, which measures the overlap between predicted and ground truth
+bounding boxes. Together, these loss functions help optimize the model for object
+detection by minimizing the difference between the predicted and ground truth class
+probabilities and bounding boxes.
+
+
+"""
+
+optimizer = tf.keras.optimizers.Adam(
+    learning_rate=LEARNING_RATE,
+    global_clipnorm=GLOBAL_CLIPNORM,
+)
+
+yolo.compile(
+    optimizer=optimizer, classification_loss="binary_crossentropy", box_loss="iou"
+)
+
+"""
+## COCO Metric Callback
+
+We will be using `BoxCOCOMetrics` from KerasCV to evaluate the model and calculate the
+Map(Mean Average Precision) score, Recall and Precision.
+"""
+
+
+class EvaluateCOCOMetricsCallback(keras.callbacks.Callback):
+    def __init__(self, data):
+        super().__init__()
+        self.data = data
+        self.metrics = keras_cv.metrics.BoxCOCOMetrics(
```

Comment:
Could it be used directly in model.compile? No.ooo.oo

Reply:
Nope, yolov8 does not support user-defined metric yet!